### PR TITLE
DEVPROD-4073: make spawn host stop/start jobs more resilient to interruptions and errors

### PR DESCRIPTION
**Pull request by Kimchelly**
DEVPROD-4073

### Description
The spawn host stop/start need to be more reliable than they currently are for sleep schedules to reliably stop/start hosts. In particular, we need them to work if the host ends up in an unusual state due to app server interruptions and timeouts. For example, if a job is stopping the spawn host but EC2 hasn't said the instance is stopped yet, it could get interrupted by a deploy restarting the app server, leaving the host in an intermediate "stopping" state. After the app server restarts, the job should eventually re-run to try stopping the host again, until EC2 reports the host is stopped (i.e. the host should eventually change from "stopping" to "stopped"). Currently, the status checks prevent re-stopping a host that's "stopping", so I fixed it to be more permissive.

In addition, the jobs should re-run again soon after the app server restarts (they currently wait 10+ mins before re-running). To achieve this, I moved the jobs to their own Amboy queue (so they have less contention to run) and will set some admin settings to speed up the re-run process.

* Make spawn host stop/start more resilient against errors and idempotent for edge cases where the operation is interrupted or the spawn host is already in its goal state (e.g. it's already stopped and the spawn host stop job tries to stop it again).
* Put spawn host modification jobs in their own Amboy queue to dispatch them faster.

### Testing
* Added more unit tests.
* Tested in staging by manually setting some DB state to simulate the edge cases, and then pressing the stop/start buttons in Spruce.
* Manually verified that EC2 API is idempotent for stop/start. In other words, stopping an already-stopped host just returns success, and starting and already-running host just returns success.

### Documentation
N/A

### CODE REVIEW

File: globals.go [hunk 1]
```diff
@@ -902,6 +902,22 @@ var (
 		HostStopped,
 	}
 
+	// StoppableHostStatuses represent all host statuses when it is possible to
```
Comment (ablack12) [inline globals.go:905]: I think the PR explains well why these might get in wonky statuses, and I can think of reasons why it'd happen, like if the user decided to start the instance early we wouldn't want to "fail" to start it. Could we add some detail here to explain? Just looking at these lists, it seems like an accident that "stopped" is in stoppable and "running" is in startable.

**Review by ablack12 — APPROVED** [review]

File: globals.go (hunk 1 above)
Comment (Kimchelly) [reply globals.go:905]: Done, added some comments for each specific state.
